### DISCUSSION TITLE
Edge Function Deploy DNS Error

### DISCUSSION THREAD

**Discussion by wchoi6** [supabase/supabase#27833] [Questions]
I'm trying to deploy edge functions (brand new to this) and keep running into DNS error issues.

I'll have imports such as 
`import { createClient } from "jsr:@supabase/supabase-js@2";`

and will get an error like this when I run `supbase functions deploy <name-of-function>`:

> DEBUG Import 'https://jsr.io/@supabase/supabase-js/meta.json' failed: error sending request for url (https://jsr.io/@supabase/supabase-js/meta.json): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution. Retrying...

This happens across different imports from various sources (esm.sh, deno.land etc).

The only solution I've found is to package the imports in a separate .ts file and then use deno bundle to generate a bundle.js file that I then use to import the necessary packages but this seems very inefficient and now I'm hitting a wall with the above import in particular because it doesn't seem to work with 'deno bundle'. 

So this may be an incredibly n00b question, but is there a setting I haven't found yet that is preventing the DNS from finding the links correctly?

**Comment by timrutter** [comment]
yep me too... starting to consider AWS lambda instead.  This should be easy

**Reply by wchoi6** [reply]
For whatever reason, the Docker instance will just run into this issue, and I've found that restarting Docker will fix it generally. Sometimes, I can just re-run the deploy command and after 2 or 3 tries it works (usually this is after I change locations and come in from a different network, but it's not always the case)

**Comment by winzamark123** [comment]
+1 sometimes getting this when edge function is invoked.
